### PR TITLE
Parse id arguments as their mentions where appropriate

### LINKED ISSUE #274: Cannot use *Id arguments like the full counterparts in prefix commands
The message differs between `current` and `serenity-next` (I haven't tested next here) but you can't use it like the below or the full discord.com link. The full counterparts do support this.

image

### PR DESCRIPTION
**Pull request by jamesbt365**
Fixes #274.

### CODE REVIEW

File: src/prefix_argument/mod.rs [hunk 1]
```diff
@@ -125,6 +125,45 @@ impl std::fmt::Display for InvalidBool {
...
 
+/// Error thrown when the user enters a string that cannot be parsed as a UserId.
+#[derive(Default, Debug)]
+pub struct InvalidUserId {
+    #[doc(hidden)]
+    pub __non_exhaustive: (),
+}
+impl std::fmt::Display for InvalidUserId {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.write_str("Entire a valid user's ID or a mention.")
```
Comment (GnomedDev) [inline src/prefix_argument/mod.rs]: Spelling?

Comment (jamesbt365) [reply src/prefix_argument/mod.rs]: Shit

File: src/prefix_argument/argument_trait.rs [hunk 2]
```diff
@@ -135,3 +138,84 @@ impl<'a> PopArgumentHack<'a, serenity::Attachment> for &PhantomData<serenity::At
...
+            .parse()
+            .ok()
+            .or_else(|| serenity::utils::parse_channel_mention(&string))
+        {
+            Ok((args.trim_start(), attachment_index, user_id))
+        } else {
+            Err((InvalidChannelId::default().into(), Some(string)))
+        }
+    }
+}
```
Comment (GnomedDev) [inline src/prefix_argument/argument_trait.rs]: This is super repetitive, can you just turn it into a macro?

**Review by GnomedDev — APPROVED** [review]
Looks perfect now!

**Pull request merged**